akka-actor/src/main/scala/akka/actor/dungeon/FaultHandling.scala

```diff
@@ -44,11 +57,37 @@ private[akka] trait FaultHandling { this: ActorCell =>
    * a restart with dying children)
    * might well be replaced by ref to a Cancellable in the future (see #2299)
    */
-  private var _failed: ActorRef = null
-  private def isFailed: Boolean = _failed != null
-  private def setFailed(perpetrator: ActorRef): Unit = _failed = perpetrator
-  private def clearFailed(): Unit = _failed = null
-  private def perpetrator: ActorRef = _failed
+  private var _failed: FailedInfo = NoFailedInfo
```

Comment:
This is nice and only adds a bit of added memory during the failed state which should be ok.